cotainr/pack.py

```diff
@@ -249,3 +316,147 @@ def _download_miniforge_installer(self, *, installer_path):
 
         else:
             raise url_error
+
+    def _run_command_in_sandbox(self, *, cmd):
+        """
+        Wrap the sandbox command runner to use class specific log_dispatcher.
+
+        Wraps calls to `self.sandbox.run_command_in_container` to use
+        self.log_dispatcher for log handling instead of the sandbox's log
+        dispatcher.
+
+        Parameters
+        ----------
+        cmd : str
+            The command to run in the container sandbox.
+
+        Returns
+        -------
+        process : :class:`subprocess.CompletedProcess`
+            Information about the process that ran in the container sandbox.
+        """
+        return self.sandbox.run_command_in_container(
+            cmd=cmd, custom_log_dispatcher=self.log_dispatcher
+        )
+
+    @property
+    def _conda_verbosity_arg(self):
+        """
+        Get a verbosity level for Conda commands.
+
+        A mapping of the internal cotainr verbosity level to `Conda verbosity
+        flags
+        <https://docs.conda.io/projects/conda/en/latest/commands/create.html#Output,%20Prompt,%20and%20Flow%20Control%20Options>`_.
+
+        Returns
+        -------
+        verbosity_arg : str
+            The verbosity arg ("-q", "-v", "-vv", etc.) to add to the Conda
+            command.
+        """
+        if self._verbosity <= 0:
+            return " -q"
+        elif self._verbosity == 2:
+            # Conda INFO
+            return " -v"
+        elif self._verbosity == 3:
+            # Conda DEBUG
+            return " -vv"
+        elif self._verbosity >= 4:
+            # Conda TRACE
+            return " -vvv"
+        else:
+            return ""
+
+    @property
+    def _logging_filters(self):
+        """
+        Create logging filters for messages from conda commands.
+
+        Returns
+        -------
+        logging_filters : list of :class:`logging.Filter`.
+            The list of filters to use with the logging machinery when handling
+            messages from conda commands.
+        """
+
+        class StripANSIEscapeCodes(logging.Filter):
+            """
+            In-place strip all ANSI escape codes.
+
+            Regex from https://stackoverflow.com/a/14693789
+            """
+
+            ansi_escape_re = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
```

Comment:
all regex needs a comment on what we are trying to match. Even if we link to stackoverflow. Not a blocker

Reply:
I will add a task to the backlog to properly document regex's with examples and a humanly readable template of what we are trying to match.